.github/workflows/mercury_bot.yml

```
@@ -33,39 +33,14 @@ jobs:
         with:
           python-version: ${{ steps.setup-python.outputs.python-version }}
 
-      - name: get files changed
-        id: get_files_changed
-        env:
-          BOT_TOKEN: ${{ secrets.BOT_TOKEN }}
-          PR_API_URL: ${{ github.event.pull_request._links.self.href }}
-        run: |
-          # get files in PR
-          ./ve1/bin/pr-artifact --api-url="${PR_API_URL}" \
-                                --get-files
-
       - name: check if only an OWNERS file is pushed
         id: check_for_owners
         env:
-          PR_FILES: ${{ steps.get_files_changed.outputs.pr_files }}
+          BOT_TOKEN: ${{ secrets.BOT_TOKEN }}
+          PR_API_URL: ${{ github.event.pull_request._links.self.href }}
         run: |
-          # check if PR contains just one partner OWNERS file
-          first_file=$(yq '.0' <<< "${PR_FILES}")
-          total_file_count=$(yq 'length' <<< ${PR_FILES})
-
-          if [ "${total_file_count}" == 1 ]; then
-            if [[ "${first_file}" == "charts/partners/"*/*"/OWNERS" ]] ; then
-                echo "An OWNERS file has been modified or added"
-                echo "merge_pr=true" | tee -a $GITHUB_OUTPUT
-            else
-              echo "The file in the PR is not a partner OWNERS file"
-              echo "merge_pr=false" | tee -a $GITHUB_OUTPUT
-              echo "msg=ERROR: PR does not include a partner OWNERS file." >> $GITHUB_OUTPUT
-            fi
-          else
-            echo "The PR contains multiple files."
-            echo "msg=ERROR: PR contains multiple files." >> $GITHUB_OUTPUT
-            echo "merge_pr=false" | tee -a $GITHUB_OUTPUT
-          fi
+          ./ve1/bin/check-for-owners --api-url="${PR_API_URL}" \
+            --allowed-category partner
```

Whoops! One more thing: This should be:
```suggestion
            --allowed-category partners
```